### ISSUE TITLE
HierarchyRequestError

### ISSUE THREAD

**Issue by monolithed** [amannn/next-intl#944]
### Description

1. At first, I followed the [tutorial](https://next-intl-docs.vercel.app/docs/getting-started/app-router#page) exactly as written and encountered the following error:

```
Error: Failed to call `useTranslations` because the context from `NextIntlClientProvider` was not found.
```

2. Then, I found an [example](https://github.com/amannn/next-intl/blob/35069bac25c03cd6fd10c802fcd8627e3eac75dd/examples/example-app-router/src/app/%5Blocale%5D/page.tsx
) of "proper implementation", but it resulted in the same error as above. Moreover, it confused me even more because I found an issue stating that we should avoid using `unstable_setRequestLocale`.

3. I added `NextIntlClientProvider` however, when I call `useMessages()`, I get a 404 error.

In summary, no matter where I use `useTranslations` or `useMessages`, I always encounter an error. I'm confused.

### Mandatory reproduction URL

https://github.com/test-examples/next-intl

**Comment by amannn** [comment]
Static rendering should not be implemented via `force-static` (please see [the corresponding docs](https://next-intl-docs.vercel.app/docs/routing/middleware#unable-to-find-locale)). Furthermore, `unstable_setRequestLocale` should be called before you call hooks like `useMessages` or `useTranslations` (see [the corresponding docs](https://next-intl-docs.vercel.app/docs/getting-started/app-router#add-unstable_setrequestlocale-to-all-layouts-and-pages)).

There's unfortunately a bit of config involved to get static rendering working correctly, but hopefully we can [get rid of this in the future](https://github.com/amannn/next-intl/issues/663).

If case you don't need static rendering, you can also avoid this API.
